src/Plugin/WebformElement/NetsEasyPaymentElement.php

```diff
@@ -126,8 +136,8 @@ public function alterForm(array &$element, array &$form, FormStateInterface $for
     $webformCurrentPage = $form['progress']['#current_page'];
     // Check if we are on the preview page.
     if ($webformCurrentPage === "webform_preview") {
-
       $amountToPay = $this->paymentHelper->getAmountToPay($formState->getUserInput(), $this->getElementProperty($element, 'amount_to_pay'));
+      $this->tempStore->set(self::AMOUNT_TO_PAY, $amountToPay);
```

Comment:
You should use your `PaymentHelper` for storing temporary data, i.e. add a function, `setAmountToPay`, say, on the helper to keep every related to validation inside the helper.